test/integration/cartridge_hotreload_test.lua

```diff
@@ -198,3 +199,34 @@ g.test_cartridge_hotreload_labels_from_config_and_set_labels = function()
         t.assert_equals(obs.label_pairs.alias, 'main')
     end
 end
+
+g.test_cartridge_hotreload_not_reset_collectors = function()
+    local main_server = g.cluster:server('main')
+    upload_config()
+
+    main_server:exec(function()
+        local metrics = require('cartridge').service_get('metrics')
+        metrics.gauge('hotreload_checker'):set(1)
+    end)
+
+    local resp = main_server:http_request('get', '/new-metrics')
+    t.assert_equals(resp.status, 200)
```

Comment:
Either set `{raise = false}` in http_request above or remove this check. Also it would be nice if both requests were consistent (both raise errors or bot not raise errors).